frontend/src/components/Inquiries/AddInquiry.tsx

```diff
@@ -0,0 +1,118 @@
+import {
+    Button,
+    FormControl,
+    FormErrorMessage,
+    FormLabel,
+    Modal,
+    ModalBody,
+    ModalCloseButton,
+    ModalContent,
+    ModalFooter,
+    ModalHeader,
+    ModalOverlay, Textarea,
+} from "@chakra-ui/react"
+import {useMutation, useQueryClient} from "@tanstack/react-query"
+import {type SubmitHandler, useForm} from "react-hook-form"
+
+import {type ApiError, type InquiryCreate, InquiriesService} from "../../client"
+import useCustomToast from "../../hooks/useCustomToast"
+import {handleError} from "../../utils"
+
+interface AddInquiryProps {
+    isOpen: boolean
+    onClose: () => void
+}
+
+const MIN_LENGTH = 10;
+const MAX_LENGTH = 255;
+
+const AddInquiry = ({isOpen, onClose}: AddInquiryProps) => {
+    const queryClient = useQueryClient()
+    const showToast = useCustomToast()
+    const {
+        register,
+        handleSubmit,
+        reset,
+        formState: {errors, isSubmitting},
+    } = useForm<InquiryCreate>({
+        mode: "onBlur",
+        criteriaMode: "all",
+        defaultValues: {
+            text: "",
+        },
+    })
+
+    const mutation = useMutation({
+        mutationFn: (data: InquiryCreate) =>
+            InquiriesService.createInquiry({requestBody: data}),
+        onSuccess: () => {
+            showToast("Success!", "Inquiry created successfully.", "success")
+            reset()
+            onClose()
+        },
+        onError: (err: ApiError) => {
+            handleError(err, showToast)
+        },
+        onSettled: async () => {
+            await queryClient.invalidateQueries({queryKey: ["Inquiry"]})
+        },
+    })
+
+    const onSubmit: SubmitHandler<InquiryCreate> = (data) => {
+        mutation.mutate(data)
+    }
+
+    return (
+        <>
+            <Modal
+                isOpen={isOpen}
+                onClose={onClose}
+                size={{base: "sm", md: "md"}}
+                isCentered
+            >
+                <ModalOverlay/>
+                <ModalContent as="form" onSubmit={handleSubmit(onSubmit)}>
+                    <ModalHeader>Add Inquiry</ModalHeader>
+                    <ModalCloseButton/>
+                    <ModalBody pb={6}>
+                        <FormControl isRequired isInvalid={!!errors.text}>
```

Comment:
just wondering, should it be "error" or "errors?" Also, would it be safer to do errors?.text ?